### DISCUSSION TITLE
SpanPoint

### DISCUSSION THREAD

**Discussion by felix6262** [HomeSpan/HomeSpan#816] [Q&A]
Is there an example how to use the SpanPoint Class? Tried some stuff with the homespan.h on my own, but didnt work. Could need some help, because the esp thats getting the homespan also needs to comunicate by wifi with another esp.

**Comment by HomeSpan** [comment]
There is no need to look through any of the code.  There are *dozens* of examples, detailed tutorials, etc., on almost every aspect of HomeSpan (and even HomeKit itself), including dedicated pages and tutorials on SpanPoint.  Everything is accessible directly from this repo - please start on the main README page and explore from there.

**Comment by felix6262** [comment]
Ah im sorry, maybe i didnt express myself well. I used the HomeSpan stuff, but tried to implement myself the SpanPoint, and actually i didnt do wrtong as much as i thought, Anyway i found the example with main device and remote device. But i've got a question: Programming structure in remoteDevice.ino is slightly different to remoteTempsensor.ino. Why did you use the "Diagnosticmode"? And my secound question is; Why do both esp32 need the macAdress of the other one? I thought only the sending ESP32 need the Macadress of the receiving one.

**Comment by HomeSpan** [comment]
mac addresses are needed for encryption and to ensure symmetry of the class when both sending and receiving. 

not sure what you mean by diagnostic mode.

**Comment by felix6262** [comment]
i mean this line #define DIAGNOSTIC_MODE in your RemoteTempSensor.ino

In the if-Statements ask if this is defined. But it is anyway. So either im missing something or its unnecessary. However thanks for answering me. Have a great day!
